src/eredis_client.erl

```diff
@@ -519,20 +555,28 @@ maybe_reconnect(Reason, #state{queue = Queue} = State) ->
 
 %% @doc: Loop until a connection can be established, this includes
 %% successfully issuing the auth and select calls. When we have a
-%% connection, give the socket to the redis client.
-reconnect_loop(Client, #state{reconnect_sleep=ReconnectSleep, transport=Transport}=State) ->
+%% connection, send the socket to Client in a message on the form
+%% `{connection_ready, Socket}'.
+reconnect_loop(Client, ReconnectSleep, Host, Port, SocketOptions,
+               TlsOptions, ConnectTimeout, AuthCmd, Db) ->
     receive
         {'EXIT', Client, Reason} -> exit(Reason)
     after
         ReconnectSleep ->
-            case connect(State) of
```

Comment:
I think it can be a problem to sleep *before* trying to connect in some situations. For the initial connection, it's correct, but for example when the socket has been closed by the server or a TCP error, we shouldn't wait before reconnecting. We should reconnect immediately.

This code is already used by eredis, but it's changed behaviour now for eredis_sub.

Reply:
#51